### PR TITLE
[TEVA-1503] Part 1: Sign in page with a single error message

### PR DESCRIPTION
**Pull request by david-mears-dfe**
## Jira ticket URL

https://dfedigital.atlassian.net/browse/TEVA-1503

## Questions

Can you think of a neater way to customize Devise's error notification into a GovUK error summary?

## Changes in this PR:

- In order to reduce code complexity by avoiding extending/customizing Devise too heavily, we agreed with UX only to have a single, generic error message for all jobseeker sign in errors.
- The text of the error message 'Invalid email or password' has yet to be written by Content Design
- Fix the font size for page headings on various jobseeker pages
- Put the form error summaries between the title and the description on various jobseeker pages by request of UX

## Screenshots of UI changes:

Screenshot 2020-12-08 at 11 59 33

Screenshot 2020-12-08 at 12 02 53

### CODE REVIEW

File: config/locales/devise.en.yml [hunk 1]
```diff
@@ -7,10 +7,10 @@ en:
     failure:
       already_authenticated: You are already signed in.
       inactive: Your account is not activated yet.
-      invalid: Invalid %{authentication_keys} or password.
+      invalid: Invalid email or password
```
Comment (david-mears-dfe) [inline config/locales/devise.en.yml:10]: Made un-dynamic because dynamic version capitalizes 'Email'

**Review by alexbowen — APPROVED** [review]
LGTM

File: app/controllers/jobseekers/sessions_controller.rb [hunk 2]
```diff
@@ -1,4 +1,15 @@
 class Jobseekers::SessionsController < Devise::SessionsController
   after_action :replace_devise_notice_flash_with_success!, only: %i[create destroy]
   before_action :sign_out_publisher!, only: %i[create]
+
+  def new
+    if [I18n.t("devise.failure.invalid"), I18n.t("devise.failure.not_found_in_database")].include? flash[:alert]
+      self.resource = resource_class.new(sign_in_params)
+      resource.errors.add(:email, flash[:alert])
```
Comment (cpjmcquillan) [inline app/controllers/jobseekers/sessions_controller.rb:8]: If you add `flash.discard(:alert)` can you remove line 9?

File: app/controllers/jobseekers/sessions_controller.rb [hunk 3]
```diff
@@ -1,4 +1,15 @@
...
+      flash.clear
+      render :new and return
```
Comment (cpjmcquillan) [inline app/controllers/jobseekers/sessions_controller.rb:10]: And why do we need the return here?

File: app/controllers/jobseekers/sessions_controller.rb (hunk 2 above)
Comment (david-mears-dfe) [reply app/controllers/jobseekers/sessions_controller.rb:8]: I can't seem to make flash.discard(:something) change flash in any way :(

File: app/views/jobseekers/registrations/new.html.haml [hunk 4]
```diff
@@ -2,20 +2,20 @@
...
   .govuk-grid-column-two-thirds
-    %h1.govuk-heading-l= t(".title")
-
-    %p.govuk-body= t(".description")
+    %h1.govuk-heading-xl= t(".title")
 
     = form_for resource, url: jobseeker_registration_path do |f|
       = f.govuk_error_summary
 
+      %p.govuk-body= t(".description")
```
Comment (cpjmcquillan) [inline app/views/jobseekers/registrations/new.html.haml:10]: I don't think the descriptions should be inside the form?

File: app/controllers/jobseekers/sessions_controller.rb (hunk 2 above)
Comment (cpjmcquillan) [reply app/controllers/jobseekers/sessions_controller.rb:8]: Don't worry then 😂

File: app/views/jobseekers/registrations/new.html.haml (hunk 4 above)
Comment (david-mears-dfe) [reply app/views/jobseekers/registrations/new.html.haml:10]: See commit message

Comment (cpjmcquillan) [reply app/views/jobseekers/registrations/new.html.haml:10]: Hmm ok I concede defeat, just looked a bit weird to me!

Comment (alexbowen) [reply app/views/jobseekers/registrations/new.html.haml:10]: i dont think its against the form gods to have textual elements inside a form except they'd usually be label elements

**Comment by david-mears-dfe** [comment]
This is closing because Content believes it to be a high priority for us to do some extension of Devise in order to make this very frequently used form more user-friendly. Namely, I'm going to create a SignUpForm object to handle some validations on blank fields and email format (but not password length). Significantly different code, so a different PR.

**Pull request closed without merging**